src/averageAges.js

```diff
@@ -37,7 +38,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = people.filter(person => person.sex === 'f');
+
+  const mothers = people.map(
+    person => person.mother).filter(mother => mother !== null);
+
+  const isWithChildren = withChildren ? people.filter(
+    person => mothers.includes(person.name)) : women;
+
+  return isWithChildren.reduce((sumOfAges, mother) => sumOfAges + (
+    mother.died - mother.born), 0) / isWithChildren.length;
```

Comment:
a lot of iteration. can be simplier
```suggestion
const women = people
    .filter(({ sex, name }) => withChildren
      ? sex === 'f' && people.some(({ mother }) => name === mother)
      : sex === 'f');

  return women.reduce((sumOfAges, mother) => sumOfAges + (
    mother.died - mother.born), 0) / women.length;
```